### PR TITLE
signin and signup api created

### PR DESCRIPTION
**Pull request by jay-m-simformsolutions**

### CODE REVIEW

**Comment by jay-m-simformsolutions** [comment]
@abhishek-simform  please review this commited code

**Comment by abhi5658** [comment]
@jay-m-simformsolutions  For every route level function we can use **async** keyword with function so that we don't need to chain the function call & save us from callback hell.
```javascript
labRoute.post('/signup', (req,res) => {

    const text = 'INSERT INTO laboratory VALUES($1, $2, $3) RETURNING *'
    const values = [req.body.labID, req.body.labPass, req.body.labName]

    getHashedPass(req.body.labPass).then(pass => {
        values[1] = pass
        client
            .query(text, values)
            .then(resp => {
                delete resp.rows[0].labPass
                res.status(200).send(resp.rows[0])
            })
            .catch(e => {
                res.status(403).send('Something went wrong! Please try again')
                console.error(e.stack)
            })
    })
})
```

can be converted to 
```javascript
// async function used - which will allow us to use await inside function
labRoute.post('/signup', async (req,res) => {
    try {
        const text = 'INSERT INTO laboratory VALUES($1, $2, $3) RETURNING *'
        const values = [req.body.labID, req.body.labPass, req.body.labName]

        // you can combine below 2 statement into 1
        const pass = await getHashedPass(req.body.labPass)
        values[1] = pass;

        // executing query
        const resp = await client.query(text, values);
        
        // deleting password hash
        delete resp.rows[0].labPass;
        
        //sending success response (can also use .json(resp.rows[0]) instead of .send )
        res.status(200).send(resp.rows[0]);
    } catch (e => {
        // status code should be 500 for server side error
        res.status(500).send('Something went wrong! Please try again')
        console.error(e.stack)
    });
});
```

**Review by abhishek-simform — APPROVED** [review]
Everything looks good.

File: src/Controllers/auth.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    const authHeader = req.headers['authorization']
+    const token = authHeader.split(' ')[1]
+
+    jwt.verify(token, process.env.JWT_SECRET, async (err,user) => {
+        if(err){
+            console.log(user);
+            console.log(err);
+            res.status(403).json({ message : 'You are not authorized', errors: err })
+        }
+        switch(user.role) {
```
Comment (abhishek-simform) [inline src/Controllers/auth.js]: Transfer this switch logic to a separate function and use it as a middleware
**verifyToken** function's only task should be to verify token

Comment (jay-m-simformsolutions) [reply src/Controllers/auth.js]: Done sir!

**Pull request closed without merging**